{
  "path": "/documentation/localauthentication/laerror/code",
  "type": "Enumeration",
  "name": "LAError.Code",
  "desc": "Error codes that the framework returns when policy evaluation fails.",
  "items": [
    {
      "name": "case appCancel",
      "desc": "The app canceled authentication."
    },
    {
      "name": "case systemCancel",
      "desc": "The system canceled authentication."
    },
    {
      "name": "case userCancel",
      "desc": "The user tapped the cancel button in the authentication dialog."
    },
    {
      "name": "case biometryDisconnected",
      "desc": "The device supports biometry only using a removable accessory, but the paired accessory isn’t connected."
    },
    {
      "name": "case biometryNotPaired",
      "desc": "The device supports biometry only using a removable accessory, but no accessory is paired."
    },
    {
      "name": "static var biometryLockout: LAError.Code",
      "desc": "Biometry is locked because there were too many failed attempts."
    },
    {
      "name": "static var biometryNotAvailable: LAError.Code",
      "desc": "Biometry is not available on the device."
    },
    {
      "name": "static var biometryNotEnrolled: LAError.Code",
      "desc": "The user has no enrolled biometric identities."
    },
    {
      "name": "case authenticationFailed",
      "desc": "The user failed to provide valid credentials."
    },
    {
      "name": "case invalidContext",
      "desc": "The context was previously invalidated."
    },
    {
      "name": "case notInteractive",
      "desc": "Displaying the required authentication user interface is forbidden."
    },
    {
      "name": "case passcodeNotSet",
      "desc": "A passcode isn’t set on the device."
    },
    {
      "name": "case userFallback",
      "desc": "The user tapped the fallback button in the authentication dialog, but no fallback is available for the authentication policy."
    },
    {
      "name": "case watchNotAvailable",
      "desc": "An attempt to authenticate with Apple Watch failed."
    },
    {
      "name": "var kLAErrorDomain: String",
      "desc": "The error domain used by LocalAuthentication."
    },
    {
      "name": "var kLAErrorAppCancel: Int32",
      "desc": "The app canceled authentication."
    },
    {
      "name": "var kLAErrorSystemCancel: Int32",
      "desc": "The system canceled authentication."
    },
    {
      "name": "var kLAErrorUserCancel: Int32",
      "desc": "The user tapped the cancel button in the authentication dialog."
    },
    {
      "name": "var kLAErrorBiometryDisconnected: Int32",
      "desc": "The device supports biometry only using a removable accessory, but the paired accessory isn’t connected."
    },
    {
      "name": "var kLAErrorBiometryLockout: Int32",
      "desc": "Biometry is locked because there were too many failed attempts."
    },
    {
      "name": "var kLAErrorBiometryNotAvailable: Int32",
      "desc": "Biometry is not available on the device."
    },
    {
      "name": "var kLAErrorBiometryNotEnrolled: Int32",
      "desc": "The user has no enrolled biometric identities."
    },
    {
      "name": "var kLAErrorBiometryNotPaired: Int32",
      "desc": "The device supports biometry only using a removable accessory, but no accessory is paired."
    },
    {
      "name": "var kLAErrorTouchIDLockout: Int32",
      "desc": "Touch ID is locked because there were too many failed attempts."
    },
    {
      "name": "var kLAErrorTouchIDNotAvailable: Int32",
      "desc": "Touch ID is not available on the device."
    },
    {
      "name": "var kLAErrorTouchIDNotEnrolled: Int32",
      "desc": "The user has no enrolled Touch ID fingers."
    },
    {
      "name": "var kLAErrorAuthenticationFailed: Int32",
      "desc": "The user failed to provide valid credentials."
    },
    {
      "name": "var kLAErrorInvalidContext: Int32",
      "desc": "The context was previously invalidated."
    },
    {
      "name": "var kLAErrorNotInteractive: Int32",
      "desc": "Displaying the required authentication user interface is forbidden."
    },
    {
      "name": "var kLAErrorPasscodeNotSet: Int32",
      "desc": "A passcode isn’t set on the device."
    },
    {
      "name": "var kLAErrorUserFallback: Int32",
      "desc": "The user tapped the fallback button in the authentication dialog, but no fallback is available for the authentication policy."
    },
    {
      "name": "var kLAErrorWatchNotAvailable: Int32",
      "desc": "An attempt to authenticate with Apple Watch failed."
    },
    {
      "name": "case invalidDimensions"
    }
  ],
  "declaration": "enum LAError.Code : Int, _ErrorCodeProtocol, @unchecked Sendable",
  "conformsTo": [
    "Sendable"
  ]
}